### ISSUE TITLE
Wrong merging of MNVs

### ISSUE THREAD

**Issue by MagdalenaZZ** [samtools/bcftools#845]
I have two files on which I do:

bcftools merge

The first one looks like this:
chr2	79687519	.	GG	TT	.	PASS	

The second one looks like this:
chr2	79687519	.	G	T	.	PASS	
chr2	79687520	.	G	T	.	PASS	

The merged file looks like this:
chr2	79687519	.	GG	TT,TG	.	PASS     ...... 0/1    0/2

So the merged file is clearly erroneous, making an arbitrary "TG" which does not exist in any of the input sample files.
I have checked, and I have several examples all showing the same error.

bcftools --version
bcftools 1.8
Using htslib 1.8
Copyright (C) 2016 Genome Research Ltd.

I apologise I cannot submit an example file, but you should find this happens consistently in each/any file you try.

**Comment by pd3** [comment]
I don't think this is an error - isn't `79687519: GG>TG` identical to `79687519: G>T`?

**Comment by MagdalenaZZ** [comment]
No pd3, look again
it is 
G > T   (79687519)
G > T   (79687520)
Should be GG > TT

**Comment by pd3** [comment]
The example needs to show also the samples, not just the variants. This is the current behavior (1.8, 1.9):
````
sample A:
 934180 CC>GG 0/1

sample B: 
 934180 C>G 0/1 
 934181 C>G 0/1

merged A B:
  934180 CC>GG,GC 0/1 0/2
  934181 C>G      ./. 0/1
`````
Although there is a logic in it, it is not the correct and you are absolutely right that the program does not produce the best possible output which should be:
````
  934180 CC>GG,GC 0/1 0/2
````

This is a known limitation. In order to merge MNVs correctly, one needs to normalize both files and then merge. Unfortunately, the MNV normalization does not work well either. This is one of the most desired unimplemented feature. (A related issue #128.)

**Comment by MagdalenaZZ** [comment]
Thanks pd3! 
I'll just use vt to atomise the MNVs in all files and then progress, so I can do a workaround that works for me. But I'm glad the issue is on your radar, so maybe in future we get some improvements.  Known issue are better than unknown issues, isn't it ;-)
